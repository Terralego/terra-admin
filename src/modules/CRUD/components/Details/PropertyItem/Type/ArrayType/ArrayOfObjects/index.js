import ArrayOfObjects from './ArrayOfObjects';

export default ArrayOfObjects;
