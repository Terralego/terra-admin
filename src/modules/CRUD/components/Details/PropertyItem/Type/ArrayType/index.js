import ArrayType from './ArrayType';

export default ArrayType;
