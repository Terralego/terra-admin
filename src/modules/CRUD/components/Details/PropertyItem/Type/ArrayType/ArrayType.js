import React from 'react';
import PropTypes from 'prop-types';
import ArrayOfObjects from './ArrayOfObjects';
import NoValue from '../../NoValue';
import { isHTML } from '../Type';

const ArrayType = props => {
  const {
    display_value: displayValue,
    schema: { items: { type } },
  } = props;
  if (displayValue === null || !displayValue.length) {
    return <NoValue />;
  }

  if (type === 'object') {
    return <ArrayOfObjects {...props} />;
  }

  if (isHTML(displayValue[0])) {
    return (
      <div className="details__RTE">
        {displayValue.map((item, index) => (
          <div
            // eslint-disable-next-line react/no-array-index-key
            key={index}
            // eslint-disable-next-line react/no-danger
            dangerouslySetInnerHTML={{ __html: item }}
          />
        ))}
      </div>
    );
  }
  return displayValue.join(', ');
};

ArrayType.propTypes = {
  display_value: PropTypes.arrayOf(PropTypes.any),
  schema: PropTypes.shape({
    items: PropTypes.shape({
      type: PropTypes.string,
    }),
  }),
};

ArrayType.defaultProps = {
  display_value: null,
  schema: {
    items: {
      type: undefined,
    },
  },
};

export default ArrayType;
