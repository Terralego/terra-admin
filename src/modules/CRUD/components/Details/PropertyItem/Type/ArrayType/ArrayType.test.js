
import React from 'react';
import renderer from 'react-test-renderer';
import ArrayType from './ArrayType';

jest.mock('..', () => props => <div {...props} />);
jest.mock('../../NoValue', () => () => <div>No value</div>);
jest.mock('./ArrayOfObjects', () => props => <div {...props} />);

const arrayStringsProps = {
  display_value: ['Chemin du Puy-en-Velay', 'Chemin des Piémonts '],
  schema: {
    items: {
      enum: [
        'Chemin du Puy-en-Velay',
        "Chemin d'Arles",
        'Chemin de Tours',
        'Chemin des Piémonts ',
        'Chemin de Vézelay',
        'Autre itinéraire',
      ],
      type: 'string',
    },
  },
};

const arrayObjectsProps = {
  display_value: [{
    date: '2019',
    source: 'Requalification du site',
  }, {
    date: '2018',
    source: 'Réfection toiture',
  }],
  schema: {
    items: {
      properties: {
        date: {
          type: 'string',
          title: 'Date',
          format: 'date',
        },
        source: {
          type: 'string',
          title: 'Source',
        },
      },
      type: 'object',
    },
  },
  ui_schema: {
    items: {
      date: {
        'ui:help': 'YYYY/MM/DD format',
      },
    },
  },
};

describe('should render correctly', () => {
  it('array of strings', () => {
    const tree = renderer.create((
      <ArrayType {...arrayStringsProps} />
    )).toJSON();
    expect(tree).toMatchSnapshot();
  });
  it('array of objects', () => {
    const tree = renderer.create((
      <ArrayType
        {...arrayObjectsProps}

      />
    )).toJSON();
    expect(tree).toMatchSnapshot();
  });
  it('nothing', () => {
    const tree = renderer.create((
      <ArrayType {...arrayStringsProps} display_value={[]} />
    )).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
