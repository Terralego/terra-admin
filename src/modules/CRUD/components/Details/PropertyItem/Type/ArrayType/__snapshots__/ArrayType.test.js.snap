// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render correctly array of objects 1`] = `
<div
  display_value={
    Array [
      Object {
        "date": "2019",
        "source": "Requalification du site",
      },
      Object {
        "date": "2018",
        "source": "Réfection toiture",
      },
    ]
  }
  schema={
    Object {
      "items": Object {
        "properties": Object {
          "date": Object {
            "format": "date",
            "title": "Date",
            "type": "string",
          },
          "source": Object {
            "title": "Source",
            "type": "string",
          },
        },
        "type": "object",
      },
    }
  }
  ui_schema={
    Object {
      "items": Object {
        "date": Object {
          "ui:help": "YYYY/MM/DD format",
        },
      },
    }
  }
/>
`;

exports[`should render correctly array of strings 1`] = `"Chemin du Puy-en-Velay, Chemin des Piémonts "`;

exports[`should render correctly nothing 1`] = `
<div>
  No value
</div>
`;
